agent/structs/config_entry_gateways.go

```diff
@@ -231,11 +239,49 @@ func (e *IngressGatewayConfigEntry) validateServiceSDS(lis IngressListener, svc
 	return nil
 }
 
+func validateGatewayTLSConfig(tlsCfg GatewayTLSConfig) error {
+	if tlsCfg.TLSMinVersion != types.TLSVersionUnspecified {
+		if err := types.ValidateTLSVersion(tlsCfg.TLSMinVersion); err != nil {
+			return err
+		}
+	}
+
+	if tlsCfg.TLSMaxVersion != types.TLSVersionUnspecified {
+		if err := types.ValidateTLSVersion(tlsCfg.TLSMaxVersion); err != nil {
+			return err
+		}
+	}
+
+	// TODO: Is there any reasonable way to validate that TLSMaxVersion is
+	// greater than or equal to TLSMinVersion as string types?
+
+	if len(tlsCfg.CipherSuites) != 0 {
+		// TODO: is there any contextual way to determine Envoy's default TLS min and max versions if either
```

Comment:
No, because
1. This code is not strictly related to envoy here, it's just generic config entry stuff.
2. You don't know what envoy version it'll be used with after it's written, and the envoy defaults may change.

Reply:
Makes sense, thanks.